.circleci/test-deploy.yml

```diff
@@ -6,18 +6,30 @@ orbs:
 filters: &filters
   tags:
     only: /.*/
-
-jobs:
-  command-tests:
+executors:
+  alpine:
+    docker: 
+      - image: alpine:latest
+  docker-base:
     docker:
-      - image: cimg/base:current
-    steps:
-      - aws-health/check-health
+      - image: cimg/base:stable
+  windows:
+    machine:
+      image: windows-server-2022-gui:current
+    shell: bash.exe
+    resource_class: windows.medium
+  macos:
+    macos:
+      xcode: 13.3
 workflows:
   test-deploy:
     jobs:
-      - command-tests:
+      - aws-health/check-health:
+          name: integration-test-<<matrix.executor>>
           context: CPE_ORBS_AWS
+          matrix:
+            parameters:
+              executor: [ "alpine", "docker-base", "windows", "macos" ]
```

Comment:
would it be better to use enum here?

Reply:
enum would be correct if you're creating a parameter for a command or job and are confining the parameters to a list of choices. 

Here, I'm doing a matrix build because I want to run the same job but on 4 different executors. It's essentially a shorthand way as rewriting the same job 4 different times where the only difference is the executor.